lib/spack/spack/package_base.py

```diff
@@ -1397,7 +1397,7 @@ def fetcher(self):
         if not self.spec.versions.concrete:
             raise ValueError("Cannot retrieve fetcher for package without concrete version.")
         if not self._fetcher:
-            self._fetcher = fs.for_package_version(self)
+            self.fetcher = fs.for_package_version(self)
         return self._fetcher
```

Comment:
Do you mean?
```suggestion
            self.fetcher = fs.for_package_version(self)
        return self.fetcher
```

Reply:
No, that would introduce a circular property call, we want to return the private `self._fetcher` that is set by the public `self.fetcher` setter (that makes sure to call `fetcher.set_package`

Reply:
```suggestion
            # assign private member with the public setter api for error checking
            self.fetcher = fs.for_package_version(self)
        return self._fetcher
```

Could we document that via comment? Given the name intersection, I don't think I will be the only one seeing this code and having this question.